src/gateway_config.erl

```diff
@@ -228,10 +228,16 @@ diagnostics(Proxy) ->
                         lager:notice("Failed to get p2p status: ~p", [Error]),
                         []
                 end,
+    DiskStatus = case filelib:is_file("/tmp/disk-failure") of
```

Comment:
Could you just move this into a separate function and add it to the Base list above?